addons/mrp/models/mrp_production.py

```diff
@@ -1547,8 +1547,8 @@ def _plan_workorders(self, replan=False):
             return
 
         self.with_context(force_date=True).write({
-            'date_start': min([workorder.leave_id.date_from for workorder in workorders]),
-            'date_finished': max([workorder.leave_id.date_to for workorder in workorders])
+            'date_start': min(leave_id.date_from for leave_id in workorders.leave_id),
+            'date_finished': max(leave_id.date_to for leave_id in workorders.leave_id)
```

Comment:
I'd rather have the `filtered()` above updated to exclude workorders without `leave_id` instead.
Otherwise, if you somehow end up with all workorders that have no `leave_id`, then you'll try to write False on `date_start`, which is a field.